src/keys.cc

```diff
@@ -19,6 +19,28 @@ static Key canonicalize_ifn(Key key)
         key.modifiers = Key::Modifiers::Control;
         key.key = key.key - 1 + 'a';
     }
+
+    if (key.modifiers & Key::Modifiers::Shift)
+    {
+        if (is_basic_alpha(key.key))
+        {
+            // Shift + ASCII letters is just the uppercase letter.
+            key.modifiers &= ~Key::Modifiers::Shift;
+            key.key = to_upper(key.key);
+        }
+        else if (key.key == Key::Tab)
```

Comment:
You know what, this is a good occasion to get rid of backtab, and move the backtab == shift-tab hack into ncurses.